cineast-api/src/main/java/org/vitrivr/cineast/api/messages/lookup/BooleanLookup.java

```diff
@@ -0,0 +1,139 @@
+package org.vitrivr.cineast.api.messages.lookup;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.commons.lang3.builder.ReflectionToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.apache.commons.lang3.tuple.MutableTriple;
+import org.apache.commons.lang3.tuple.Triple;
+import org.graalvm.compiler.nodes.calc.ObjectEqualsNode;
+import org.vitrivr.cineast.api.messages.interfaces.Message;
+import org.vitrivr.cineast.api.messages.interfaces.MessageType;
+import org.vitrivr.cineast.core.data.providers.primitive.PrimitiveTypeProvider;
+import org.vitrivr.cineast.core.db.RelationalOperator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class BooleanLookup implements Message {
+
+
+    /**
+     * The type of Boolean Lookup
+     */
+
+    private String type;
+
+    /**
+     * Entity for the BooleanQuery
+     */
+    private String entity;
+    /**
+     * List consisting of single Boolean Queries
```

Comment:
This is one of those points where it would be good to have some documentation, at this point it's unclear on what the purpose of those queries is - are they subject to an `or` or an `and`?